datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/MLModelType.java

```diff
@@ -0,0 +1,115 @@
+package com.linkedin.datahub.graphql.types.mlmodel;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableSet;
+import com.linkedin.common.urn.MLModelUrn;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.generated.AutoCompleteResults;
+import com.linkedin.datahub.graphql.generated.BrowsePath;
+import com.linkedin.datahub.graphql.generated.BrowseResults;
+import com.linkedin.datahub.graphql.generated.EntityType;
+import com.linkedin.datahub.graphql.generated.FacetFilterInput;
+import com.linkedin.datahub.graphql.generated.MLModel;
+import com.linkedin.datahub.graphql.generated.SearchResults;
+import com.linkedin.datahub.graphql.resolvers.ResolverUtils;
+import com.linkedin.datahub.graphql.types.BrowsableEntityType;
+import com.linkedin.datahub.graphql.types.SearchableEntityType;
+import com.linkedin.datahub.graphql.types.mappers.AutoCompleteResultsMapper;
+import com.linkedin.datahub.graphql.types.mlmodel.mappers.MlModelMapper;
+import com.linkedin.datahub.graphql.types.mappers.SearchResultsMapper;
+import com.linkedin.metadata.query.AutoCompleteResult;
+import com.linkedin.ml.client.MLModels;
+import com.linkedin.restli.common.CollectionResponse;
+
+import lombok.NonNull;
+
+public class MLModelType implements SearchableEntityType<MLModel>, BrowsableEntityType<MLModel> {
+
+    private static final Set<String> FACET_FIELDS = ImmutableSet.of("origin", "platform");
+    private static final String DEFAULT_AUTO_COMPLETE_FIELD = "name";
+    private final MLModels _mlModelsClient;
+
+    public MLModelType(final MLModels mlModelsClient) {
+        _mlModelsClient = mlModelsClient;
+    }
+
+    @Override
+    public EntityType type() {
+        return EntityType.MLMODEL;
```

Comment:
let's make the type ML_MODEL for readbility

Reply:
Per this decision its better to leave it as `MLMODEL` - https://github.com/linkedin/datahub/pull/1896#issuecomment-780864625